signac_dashboard/views.py

```diff
@@ -65,41 +72,52 @@ def show_job(dashboard, jobid):
         return dashboard._render_job_view(default_view="grid")
 
 
-def get_file(dashboard, jobid, filename):
-    try:
-        job = dashboard.project.open_job(id=jobid)
-    except KeyError:
-        abort(404, "The job id requested could not be found.")
+def get_file(dashboard, filename, jobid=None):
+    if jobid is not None:
+        try:
+            job_or_project = dashboard.project.open_job(id=jobid)
+        except KeyError:
+            abort(404, "The job id requested could not be found.")
+        except LookupError:
+            dashboard.project.find_jobs()
+            abort(404, "Multiple jobs match the requested job id.")
     else:
-        if job.isfile(filename):
-            mimetype = None
-            cache_timeout = 0
-            # Return logs as plaintext
-            textfile_regexes = ["job-.*\\.[oe][0-9]*", ".*\\.log", ".*\\.dat"]
-            for regex in textfile_regexes:
-                if re.match(regex, filename) is not None:
-                    mimetype = "text/plain"
-            download_name = request.args.get("download_name", filename)
-            return send_from_directory(
-                directory=job.workspace(),
-                path=filename,
-                mimetype=mimetype,
-                cache_timeout=cache_timeout,
-                conditional=True,
-                download_name=download_name,
-            )
-        else:
-            abort(404, "The file requested does not exist.")
+        job_or_project = dashboard.project
+    if job_or_project.isfile(filename):
+        directory = job_or_project.fn("")
+        mimetype = None
+        cache_timeout = 0
+        download_name = request.args.get("download_name", filename)
+        return send_from_directory(
+            directory=directory,
+            path=filename,
+            mimetype=mimetype,
+            cache_timeout=cache_timeout,
+            conditional=True,
+            download_name=download_name,
+        )
+    else:
+        abort(404, "The file requested does not exist.")
 
 
 def change_modules(dashboard):
-    enabled_modules = set(session.get("enabled_modules", []))
-    for i, module in enumerate(dashboard.modules):
+    enabled_module_indices = session.get(
+        "enabled_module_indices", dict(JobContext=[], ProjectContext=[])
```

Comment:
This should probably be a `defaultdict(list)`.
```suggestion
        "enabled_module_indices", {"JobContext": [], "ProjectContext": []}
```